### ISSUE TITLE
Integration Test Failures

### ISSUE THREAD

**Issue by paulbellamy** [docker/docker-py#586]
Hey,

On 1.2.2, I am seeing 6 Test failures. Hopefully I have a misconfigured vagrant box?

Most failures seem to be to do with Binds. And one is the TestImportFromURL.

Any insight would be very appreciated. Thanks, and nice client library! Any other info you want, I'll be happy to provide.

I'm running them under vagrant, on ubuntu 14.10.
```
vagrant@vagrant-ubuntu-utopic-64:~/repos/docker-py$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 14.10
Release:        14.10
Codename:       utopic
vagrant@vagrant-ubuntu-utopic-64:~/repos/docker-py$ docker version
Client version: 1.6.0
Client API version: 1.18
Go version (client): go1.4.2
Git commit (client): 4749651
OS/Arch (client): linux/amd64
Server version: 1.6.0
Server API version: 1.18
Go version (server): go1.4.2
Git commit (server): 4749651
OS/Arch (server): linux/amd64
```

Test output is:
```
vagrant@vagrant-ubuntu-utopic-64:~/repos/docker-py$ make integration-test
docker build -t docker-py .
Sending build context to Docker daemon 557.1 kB
Sending build context to Docker daemon 621.1 kB

Sending build context to Docker daemon 
Step 0 : FROM python:2.7
 ---> d833e0b23482
Step 1 : MAINTAINER Joffrey F <joffrey@dotcloud.com>
 ---> Using cache
 ---> ee9cc62565f3
Step 2 : ADD . /home/docker-py
 ---> af75437ce099
Removing intermediate container 00f770f7928c
Step 3 : WORKDIR /home/docker-py
 ---> Running in 7e91db38cb32
 ---> 9b6f34c308ce
Removing intermediate container 7e91db38cb32
Step 4 : RUN pip install -r test-requirements.txt
 ---> Running in 8ce3ac8a6d65
Collecting mock==1.0.1 (from -r test-requirements.txt (line 1))
  Downloading mock-1.0.1.tar.gz (818kB)
Collecting coverage==3.7.1 (from -r test-requirements.txt (line 2))
  Downloading coverage-3.7.1.tar.gz (284kB)
Installing collected packages: mock, coverage
  Running setup.py install for mock
  Running setup.py install for coverage
Successfully installed coverage-3.7.1 mock-1.0.1
 ---> 9d581828897a
Removing intermediate container 8ce3ac8a6d65
Step 5 : RUN pip install .
 ---> Running in 07a9152e0287
Processing /home/docker-py
Collecting requests>=2.5.2 (from docker-py==1.2.2)
  Downloading requests-2.7.0-py2.py3-none-any.whl (470kB)
Collecting six>=1.3.0 (from docker-py==1.2.2)
  Downloading six-1.9.0-py2.py3-none-any.whl
Collecting websocket-client>=0.11.0 (from docker-py==1.2.2)
  Downloading websocket_client-0.30.0.tar.gz (191kB)
Installing collected packages: requests, six, websocket-client, docker-py
  Running setup.py install for websocket-client
  Running setup.py install for docker-py
Successfully installed docker-py-1.2.2 requests-2.7.0 six-1.9.0 websocket-client-0.30.0
 ---> 59f4c7874913
Removing intermediate container 07a9152e0287
Successfully built 59f4c7874913
docker run -v /var/run/docker.sock:/var/run/docker.sock docker-py python tests/integration_test.py
...........F....F............F...............................F.F..F.........
======================================================================
FAIL: runTest (__main__.TestCreateContainerWithBinds)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/integration_test.py", line 195, in runTest
    self.assertIn(filename, logs)
AssertionError: 'shared.txt' not found in ''

======================================================================
FAIL: runTest (__main__.TestCreateContainerWithRoBinds)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/integration_test.py", line 235, in runTest
    self.assertIn(filename, logs)
AssertionError: 'shared.txt' not found in ''

======================================================================
FAIL: runTest (__main__.TestImportFromURL)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/integration_test.py", line 1474, in runTest
    self.assertNotIn('error', result)
AssertionError: 'error' unexpectedly found in {u'errorDetail': {u'message': u'Get http://bccf739d2f13:55214: dial tcp: lookup bccf739d2f13: no such host'}, u'error': u'Get http://bccf739d2f13:55214: dial tcp: lookup bccf739d2f13: no such host'}

======================================================================
FAIL: runTest (__main__.TestStartContainerWithBinds)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/integration_test.py", line 273, in runTest
    self.assertIn(filename, logs)
AssertionError: 'shared.txt' not found in ''

======================================================================
FAIL: runTest (__main__.TestStartContainerWithFileBind)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/integration_test.py", line 342, in runTest
    self.assertEqual(exitcode, 0)
AssertionError: 1 != 0

======================================================================
FAIL: runTest (__main__.TestStartContainerWithRoBinds)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/integration_test.py", line 311, in runTest
    self.assertIn(filename, logs)
AssertionError: 'shared.txt' not found in ''

----------------------------------------------------------------------
Ran 76 tests in 152.764s

FAILED (failures=6)
Makefile:14: recipe for target 'integration-test' failed
make: *** [integration-test] Error 1
```

**Comment by shin-** [comment]
Ha, I don't use the Makefile, so I missed those. Should be an easy fix (and they should run fine if you do `python tests/integration_test.py` in your vagrant box.

I'll look into fixing, thanks for the report.

**Comment by shin-** [comment]
Fixed by #587

**Comment by paulbellamy** [comment]
Excellent! Thanks so much. :)
